server/src/main/java/io/crate/planner/optimizer/rule/EquiJoinToLookupJoin.java

```diff
@@ -75,7 +75,7 @@ public class EquiJoinToLookupJoin implements Rule<JoinPlan> {
             // only inner-equi-joins
             j.joinType() == JoinType.INNER &&
             j.joinCondition() != null &&
-            EquiJoinDetector.isEquiJoin(j.joinCondition()) &&
+            HashJoinDetector.isEquiJoin(j.joinCondition()) &&
```

Comment:
Here I need a real equi-join. This needs to be fixed. However, the rule is currently switched-off by default. But this can lead to false-positives.